src/AiClient.php

```diff
@@ -0,0 +1,373 @@
+    {
+        if (self::$defaultRegistry === null) {
+            $registry = new ProviderRegistry();
+
+            // TODO: Uncomment this once provider implementation PR #39 is merged.
+            //$registry->setHttpTransporter(HttpTransporterFactory::createTransporter());
+            //$registry->registerProvider(AnthropicProvider::class);
+            //$registry->registerProvider(GoogleProvider::class);
+            //$registry->registerProvider(OpenAiProvider::class);
+
+            self::$defaultRegistry = $registry;
+        }
+
+        return self::$defaultRegistry;
+    }
+
+    /**
+     * Sets the default provider registry instance.
+     *
+     * @since n.e.x.t
+     *
+     * @param ProviderRegistry $registry The provider registry to set as default.
+     */
+    public static function setDefaultRegistry(ProviderRegistry $registry): void
+    {
+        self::$defaultRegistry = $registry;
+    }
+
+    /**
+     * Checks if a provider is configured and available for use.
+     *
+     * @since n.e.x.t
+     *
+     * @param ProviderAvailabilityInterface $availability The provider availability instance to check.
+     * @return bool True if the provider is configured and available, false otherwise.
+     */
+    public static function isConfigured(ProviderAvailabilityInterface $availability): bool
+    {
+        return $availability->isConfigured();
+    }
+
+    /**
+     * Creates a new prompt builder for fluent API usage.
+     *
+     * This method will return an actual PromptBuilder instance once PR #49 is merged.
+     * The traditional API methods in this class will then delegate to PromptBuilder
+     * rather than implementing their own generation logic.
+     *
+     * @since n.e.x.t
+     *
+     * @param Prompt $prompt Optional initial prompt content.
+     * @return PromptBuilder The prompt builder instance.
+     */
+    public static function prompt($prompt = null): PromptBuilder
+    {
+        return new PromptBuilder(self::defaultRegistry(), $prompt);
+    }
+
+    /**
+     * Generates content using a unified API that automatically detects model capabilities.
+     *
+     * When no model is provided, this method delegates to PromptBuilder for intelligent
+     * model discovery based on prompt content and configuration. When a model is provided,
+     * it infers the capability from the model's interfaces and delegates to the capability-based method.
+     *
+     * @since n.e.x.t
+     *
+     * @param Prompt $prompt The prompt content.
+     * @param ModelInterface|null $model Optional specific model to use.
+     * @return GenerativeAiResult The generation result.
+     *
+     * @throws \InvalidArgumentException If the provided model doesn't support any known generation type.
+     * @throws \RuntimeException If no suitable model can be found for the prompt.
+     */
+    public static function generateResult($prompt, ?ModelInterface $model = null): GenerativeAiResult
```

Comment:
I am just realizing that there is a severe limitation here that we didn't consider before (this is also lacking in the `ARCHITECTURE.md` doc): We need to still allow passing a `ModelConfig`, even if we don't pass a specific `ModelInterface` instance.

I just chatted about this with @JasonTheAdams, we're thinking that we should change the second parameter here to allow `ModelInterface|ModelConfig|null`:
* It needs to remain optional.
* If the dev wants to use a specific model, they should provide that instance and set the `ModelConfig` inside of that instance already.
* Else if they don't want to use a specific model but want to use specific configuration, they should directly provide the `ModelConfig`.